src/main/assets/scripts/main.js

```diff
@@ -0,0 +1,4 @@
+// Adoption JS
```

Comment:
Does this need to be JS file? Any reason why we don't use ts consistently?

Reply:
I don't think it does need to be js. We will probably need to run the TS through a new loader of some sort but let's give it a go.